src/vm/analysis/type_checker/mod.rs

```diff
@@ -437,9 +437,16 @@ impl <'a, 'b> TypeChecker <'a, 'b> {
         if let Some(type_result) = self.try_native_function_check(function_name, args, context) {
             type_result
         } else {
-            let function_type = self.get_function_type(function_name)
-                .ok_or(CheckErrors::UnknownFunction(function_name.to_string()))?;
-            self.type_check_function_type(&function_type, args, context)
+            let function = match self.get_function_type(function_name) {
```

Comment:
Is there a reason we're not using `.ok_or()` here, like before?

Reply:
We're extracting the struct from the function_type enum.